scheduler/scheduler-examples/src/main/java/org/ow2/proactive/scheduler/examples/FTPConnector.java

```diff
@@ -0,0 +1,381 @@
+                            ex.printStackTrace();
+                        } finally {
+                            zipFile.close();
+
+                        }
+                    }
+                } // If it is a folder, download all its contents recursively
+                else {
+                    ftpClient.changeWorkingDirectory(ftpRemoteFilePath);
+                    downloadDirectory(ftpClient, "", "", ftpLocalFilePath);
+                }
+                System.out.println("END Import file(s) from FTP.");
+            }
+            //FTP mode is PUT
+            else if (ftpMode.equals("PUT")) {
+                if (ftpLocalFilePath.isEmpty()) {
+                    throw new Exception("The ftp local file path variable is empty. Please, enter a value.");
+                }
+                System.out.println("Exporting file(s) from global space at: " + ftpLocalFilePath + " to " +
+                                   ftpRemoteFilePath);
+                File localFile = new File(ftpLocalFilePath);
+                // if the ftp remote file path does not exist then it is created.
+                if (!checkDirectoryExists(ftpClient, ftpRemoteFilePath)) {
+                    ftpClient.makeDirectory(ftpRemoteFilePath);
+                    System.out.println(ftpRemoteFilePath + " does not exist. Therefore, it is created.");
+                } else {
+                    System.out.println(ftpRemoteFilePath + " exists.");
+                }
+                // If it is a single file:
+                if (localFile.isFile()) {
+                    System.out.println("A single FILE to upload");
+                    ftpClient.changeWorkingDirectory(ftpRemoteFilePath);
+                    boolean uploaded = uploadSingleFile(ftpClient, ftpLocalFilePath, localFile.getName());
+                    if (uploaded) {
+                        System.out.println("UPLOADED a file to: " + Paths.get(ftpRemoteFilePath, localFile.getName()));
+                    } else {
+                        System.out.println("COULD NOT upload the file: " + ftpLocalFilePath);
+                    }
+                }
+                // If it is a folder, upload all its contents recursively
+                else if (localFile.isDirectory()) {
+                    System.out.println("A DIRECTORY to upload");
+                    ftpClient.changeWorkingDirectory(ftpRemoteFilePath);
+                    uploadDirectory(ftpClient, "", ftpLocalFilePath, "");
+                } else {
+                    throw new Exception(localFile + " not found. Please, enter a valid path.");
+                }
+                System.out.println("END Export file(s) to FTP.");
+            }
+            // log out and disconnect from the server
+            ftpClient.logout();
+            ftpClient.disconnect();
+
+            System.out.println("Disconnected");
+        } catch (IOException ex) {
+            ex.printStackTrace();
```

Comment:
You should not catch an IOException and print the stack trace here, this exception should be thrown by the task